rfc/curio-api-rfc.md

```diff
@@ -0,0 +1,133 @@
+# Curio API RFC
+
+These are some suggested endpoints for collaboration between Curio and hot storage market software such as Storacha.
+
+## Piece Storage & Retrieval
+
+These endpoints are simply used for putting arbitrary blobs of bytes that has to a PieceCID of any valid size (i.e. 127/128 * 2^n where n is a positive whole number)
+
+### GET /piece/{piece cid v2}
+### HEAD /piece/{piece cid v2 }
+
+These can just follow FRC-066: https://github.com/filecoin-project/FIPs/blob/master/FRCs/frc-0066.md (please include the Range parameter!)
+
+With the additional stipulation that they should probably ONLY accept v2 Piece CID: https://github.com/filecoin-project/FIPs/blob/master/FRCs/frc-0069.md
+
+### PUT /piece/{piece cid v2}
```

Comment:
Might require a UCAN header for auth

Reply:
Also for scalability it would be a good idea to have the main endpoint tell you where to go with the data.

Unfortunately redirecting PUT/POST with 3xx is really wonky (pretty much broken with the Go http.Client), so it might just require some two-endpoint setup.